### ISSUE TITLE
Loading embeddings error

### ISSUE THREAD

**Issue by JPetronilhoNunes** [LM-Kit/lm-kit-net-samples#5]
In the sample custom_chatbot_with_rag, when executing the function LoadEmbeddingModel(), it gives an error and the application can´t proceed.

![image](https://github.com/user-attachments/assets/22dab308-96a5-498e-9fe0-720d13e2b8c7)

**Comment by lcarrere** [comment]
Hello Joaquim,

In our latest version, we've improved error handling to provide more information about why the loading failed.

We've also added a new property, Configuration.EnableModelChecksumValidation, which allows you to enable model file integrity validation. Setting it to true will ensure that you have not downloaded a corrupted version of the model. ref: [https://docs.lm-kit.com/lm-kit-net/api/LMKit.Global.Configuration.EnableModelChecksumValidation.html](https://docs.lm-kit.com/lm-kit-net/api/LMKit.Global.Configuration.EnableModelChecksumValidation.html
)


To summarize, could you please:

1- 
Update the LM-Kit version to the latest release by either updating packages via Visual Studio or downloading the latest state of the demo repository.

2- 
Add the following line of code:


`Configuration.EnableModelChecksumValidation = true;`

Just before:

`LMKit.Licensing.LicenseManager.SetLicenseKey("");`

3- 
If the problem persists, please share a screenshot of the new exception.

Thank you!

**Comment by JPetronilhoNunes** [comment]
Hello Icarrere,

I´ve done what you suggested and executed steps 1 and 2, nevertheless I can´t still run the application successfully, although the error is different.
Here is a screenshot:

![image](https://github.com/user-attachments/assets/00ebaa39-fda9-4dfa-9d86-c0b67c9c6797)

**Comment by lcarrere** [comment]
okay thanks, this looks like a gpu related issue. 

Can you tell me the value of the backend variable produced by this snippet? Simply place it in the first lines of the app's main function.

```
LMKit.Global.Runtime.Initialize();
var backend = LMKit.Global.Runtime.Backend;
```

Also what is the produced output of this?

```
foreach (var device in LMKit.Graphics.Gpu.GpuDeviceInfo.Devices)
{
    Console.WriteLine(device.DeviceName);
    Console.WriteLine(device.DeviceDescription);
    Console.WriteLine(device.DeviceType);
}
```

**Comment by JPetronilhoNunes** [comment]
For the variable backend the value is:
backend = "Cuda12"

For each console output, this is the screenshot:
![image](https://github.com/user-attachments/assets/753b9ab2-46be-4ecc-a64b-01a3ee8ebcee)

**Comment by lcarrere** [comment]
Okay, thanks. This GPU is quite old, and there is a small chance it is not compatible with the CUDA SDK version we are using. We will try to obtain a device from this family through Amazon to see how we can handle any compatibility issues with the SDK, if any.

In the meantime, you can switch off the CUDA backend to force the toolkit to fall back to Vulkan. To do so, simply add the following line at the very start:
`LMKit.Global.Runtime.EnableCuda = false;`

Does this work?

**Comment by JPetronilhoNunes** [comment]
No, this time with a memory allocation issue:

![image](https://github.com/user-attachments/assets/17faad93-4a44-47c4-93be-edabe8006a17)

![image](https://github.com/user-attachments/assets/a9de9acc-6833-48fb-a0b2-63587498a280)

My machine:

![image](https://github.com/user-attachments/assets/f7893f02-e825-4202-b3d1-3755fa675241)

**Comment by JPetronilhoNunes** [comment]
It´s working!
I´m going to test with the other completion model you suggested.
Many thanks for your help.

**Comment by lcarrere** [comment]
> It´s working!
😄
You are very welcome Joaquim, and thank you for the return.
I will send you a notification when we have made the SDK work autonomously with your graphics card.

**Comment by lcarrere** [comment]
@JPetronilhoNunes we've finally replicated and resolved the root of the initial vector issue.

In our next release, you should be able to use your GPU with the CUDA backend without the need to precompute the number of tensors to dispatch between the GPU and CPU. This should also make your code much faster.

**Comment by JPetronilhoNunes** [comment]
Thank you for the feedback, @lcarrere!
